src/extension.ts

```diff
@@ -182,30 +174,24 @@ class ExtensionManager implements vscode.Disposable {
     telemetry.logEvent('open', telemetryProperties);
   }
 
-  public getActiveFolderContext() : StateManager | undefined {
-    const activeFolder = this.getActiveFolder();
-    if (activeFolder) {
-      const extensionState = new StateManager(this.extensionContext, activeFolder);
-      return extensionState;
-    }
-
-    return undefined;
+  public getFolderContext(folder: vscode.WorkspaceFolder): StateManager {
+    return new StateManager(this.extensionContext, folder);
   }
 
   // Partial activation means that the CMake Tools commands are hidden
   // from the commands pallette and the status bar is not visible.
-  // The context variable "cmake:enableFullFeatureSet" is always equal
-  // to the state setting ignoreCMakeListsMissing.
-  // To have them both always in sync, cmake:enableFullFeatureSet is set
-  // by the getter and setter of ignoreCMakeListsMissing.
-  public enableFullFeatureSet(fullFeatureSet: boolean) {
-    const context = this.getActiveFolderContext();
-    if (context) {
-      context.ignoreCMakeListsMissing = !fullFeatureSet;
-      this._statusBar.setVisible(fullFeatureSet);
-    } else {
-      log.trace(localize('enableFullFeatureSet.no.active.folder', 'enableFullFeatureSet({0}) called but not active folder found.'), fullFeatureSet);
-    }
+  // The context variable "cmake:enableFullFeatureSet" (which controls
+  // all the cmake commands and UI elements) is set to true,
+  // if there is at least one folder with full features set enabled.
+  // We need to add this private _enableFullFeaturesSetOnWorkspace here
+  // because currently there is no way of reading a context variable
+  // like cmake:enableFullFeatureSet, to apply the OR operation on it.
+  private _enableFullFeatureSetOnWorkspace = false;
+  public enableFullFeatureSet(fullFeatureSet: boolean, folder: vscode.WorkspaceFolder) {
+    this.getFolderContext(folder).ignoreCMakeListsMissing = !fullFeatureSet;
+    this._enableFullFeatureSetOnWorkspace = this._enableFullFeatureSetOnWorkspace || fullFeatureSet;
```

Comment:
I find this pattern confusing. You need to set `_enableFullFeatureSetOnWorkspace` to `false` outside of this method because `fullFeatureSet` can't change the true to false. Is there a reason you can't just take the value of `fullFeatureSet`?

Reply:
Do we ever need to turn support off?  I'm wondering if we should just leave the full feature set enabled once it's been enabled.

Reply:
If you remove the state that would make CMake Tools enable the full feature set, then you would need to reload the window at that point to disable it.  I would be fine with that behavior.